### PR TITLE
Fixes to multiple XR-related bugs and issues.

### PR DESCRIPTION
**Pull request by StayTalm**
Issues Fixed:
#335 #215 #211 #218 #255 #213 #250 #206 #197 #330 #207 #249 #203 #205 #201 

Added recently : #239 

Requires most up to date version of `xr/input/misc-isx-fixes` ono branch, currently changeset 7d03dfdc83ce.

Notes:
#250 : Haptics no longer has a constant rumble API for XR, and so I unhooked it from the IHaptics interface.
#201 : Until we massively update our OpenVR backend, knuckles can't work correctly, so I opted to drop it from ISX layouts.

### CODE REVIEW

File: Assets/QA/Tests/XRDeviceActions/_SharedScripts/Vector2ControlActionStatus.cs [hunk 1]
```diff
@@ -19,6 +20,22 @@ void OnEnable()
...
         vector2Action.cancelled += UpdateVector2;
+
+        ReadOnlyArray<InputControl> controls = vector2Action.controls;
+        for (int i = 0; i < controls.Count; i++)
+        {
+            Vector2Control control = controls[i] as Vector2Control;
+            if (control != null)
+            {
+                Vector2 value = control.ReadValue();
+                status1Slider.value = value.x;
```
Comment (mattdatwork) [inline Assets/QA/Tests/XRDeviceActions/_SharedScripts/Vector2ControlActionStatus.cs:31]: cant tell from the diff, but assume that the values are written back to the vector in the same order from the sliders?

File: Assets/QA/Tests/XRHaptics/XRHaptics.cs [hunk 2]
```diff
@@ -11,29 +11,19 @@ public class XRHaptics : MonoBehaviour
     public Image rightHapticDetected;
     public Image rightTryingToRumble;
 
+    public float m_RumblePeriod = 1f;
```
Comment (mattdatwork) [inline Assets/QA/Tests/XRHaptics/XRHaptics.cs:14]: should these be  defaulted to 0f?

File: Packages/com.unity.inputsystem/InputSystem/Plugins/XR/GearVR.cs [hunk 3]
```diff
@@ -4,32 +4,27 @@
 namespace UnityEngine.Experimental.Input.Plugins.XR
 {
     [InputControlLayout]
-    public class GearVRHMD : XRHMD
+    public class GearVRSimpleHMD : XRHMD
```
Comment (mattdatwork) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/XR/GearVR.cs:7]: Base rather than Simple?

**Review by mattdatwork — CHANGES_REQUESTED** [review]

**Review by jackpr-unity — CHANGES_REQUESTED** [review]
Compile warning:
D:\Git\InputSystem_StayTalmFork\Packages\com.unity.inputsystem\InputSystem\Plugins\PlayerInput\PlayerInput.cs(265,23): warning CS0108: 'PlayerInput.camera' hides inherited member 'Component.camera'. Use the new keyword if hiding was intended.

**Review still in progress:**

#203 - 3b148f9dbb4960821ee12443c104742866b098ae
Vive Tracker (SteamVR version 1.2.7)
- The Vive tracker pro has "VIVETrackerPRO" in its name, but the non-pro version's name just seems to be a serial number. Can this be fixed?

#207 - 3b148f9dbb4960821ee12443c104742866b098ae
Vive Pro
The actions not fire when the Vive Pro is being used, but not when the Vive is being used.  My Vive's name is "Vive. MV"

#211 - 3b148f9dbb4960821ee12443c104742866b098ae
GearVR/Go
- The Oculus Go still doesn't have an entry in the InputActions table.  Should it?  We know that it's equivalent to GearVR, but how are users supposed to know?

#239 
?

#255  - 3b148f9dbb4960821ee12443c104742866b098ae
Daydream
- The Remote's Tracking state is reported as 50 (0x32), but only rotation and acceleration are updating
- Otherwise OK.

**Passed:**

#197 - 3b148f9dbb4960821ee12443c104742866b098ae
OpenVR - Rift

#201 - 3b148f9dbb4960821ee12443c104742866b098ae
Knuckles
Knuckles InputActions have been removed.  This will be closed as "by design" for now

#205 - 3b148f9dbb4960821ee12443c104742866b098ae
Rift Remote

#206 - 3b148f9dbb4960821ee12443c104742866b098ae
Oculus

#213 - 3b148f9dbb4960821ee12443c104742866b098ae
Daydream

#215 - 3b148f9dbb4960821ee12443c104742866b098ae
WMR

#218 - 3b148f9dbb4960821ee12443c104742866b098ae
GearVR

#249 - 3b148f9dbb4960821ee12443c104742866b098ae
OpenVR - Vive and Rift

#250 - 3b148f9dbb4960821ee12443c104742866b098ae
Haptics

#330 - 3b148f9dbb4960821ee12443c104742866b098ae
Oculus

#335 - 3b148f9dbb4960821ee12443c104742866b098ae
WMR

File: Packages/com.unity.inputsystem/InputSystem/Plugins/XR/GearVR.cs (hunk 3 above)
Comment (jackpr-unity) [reply Packages/com.unity.inputsystem/InputSystem/Plugins/XR/GearVR.cs:7]: "Base" is probably better politically for our relationship with Oculus/Samsung

**Comment by Rene-Damm** [comment]
Sorry for my super lateness here. Getting on this now.

**Review by Rene-Damm — APPROVED** [review]
LGTM. Thinking the same as other commenters RE "simple" vs "base" but leaving that up to you guys. Will merge as is. Rename can come any time.

**Comment by Rene-Damm** [comment]
@StayTalm Forgot to mention, when making changes, please add respective change documentation to CHANGELOG.md.

**Comment by Rene-Damm** [comment]
Merged this one too early. Sorry guys, missed that you requested changes. Was a long day yesterday :)

**Pull request merged**